Frameworks/CoreGraphics/CGImage.mm

```diff
@@ -21,981 +21,660 @@
+    ComPtr<IStream> spStream;
+    RETURN_NULL_IF_FAILED(::CreateStreamOnHGlobal(NULL, FALSE, &spStream));
+
+    ULONG written = 0;
+    HRESULT hr;
+
+    hr = spStream->Write(bytes, length, &written);
+    if (FAILED(hr) || (written != length)) {
+      TraceError(TAG, L"IStream::Write failed hr=%x len=%d written=%d", hr,
+                 length, written);
+      return NULL;
+    }
+
+    // seek to start
+    LARGE_INTEGER offset;
+    offset.QuadPart = 0;
+    RETURN_NULL_IF_FAILED(spStream->Seek(offset, STREAM_SEEK_SET, NULL));
+
+    RETURN_NULL_IF_FAILED(
+        imageFactory->CreateDecoder(decoderCls, NULL, &pDecoder));
+
+    hr = pDecoder->Initialize(spStream.Get(), WICDecodeMetadataCacheOnLoad);
+    RETURN_NULL_IF_FAILED(hr);
+
+  } else {
+    ComPtr<IWICStream> spStream;
+    RETURN_NULL_IF_FAILED(imageFactory->CreateStream(&spStream));
+
+    RETURN_NULL_IF_FAILED(spStream->InitializeFromMemory(
+        static_cast<unsigned char *>(bytes), length));
+    RETURN_NULL_IF_FAILED(imageFactory->CreateDecoderFromStream(
+        spStream.Get(), nullptr, WICDecodeMetadataCacheOnDemand, &pDecoder));
+  }
+
+  ComPtr<IWICBitmapFrameDecode> bitMapFrameDecoder;
+  RETURN_NULL_IF_FAILED(pDecoder->GetFrame(0, &bitMapFrameDecoder));
+
+  CGImageRef imageRef = __CGImage::CreateInstance();
+  imageRef->ImageSource().Attach(bitMapFrameDecoder.Detach());
+  // TODO:(JJ) Set the color space
+  return imageRef;
+}
+
+NSData *_CGImagePNGRepresentation(CGImageRef image) {
+  return _CGImageRepresentation(image, GUID_WICPixelFormat32bppBGRA);
+}
+
+NSData *_CGImageJPEGRepresentation(CGImageRef image) {
+  return _CGImageRepresentation(image, GUID_WICPixelFormat32bppCMYK);
+}
+
+NSData *_CGImageRepresentation(CGImageRef image, REFGUID guid) {
+  RETURN_NULL_IF(!image);
+
+  ComPtr<IWICImagingFactory> imageFactory = _GetWICFactory();
+  ComPtr<IWICFormatConverter> imageFormatConverter;
+
+  RETURN_NULL_IF_FAILED(
+      imageFactory->CreateFormatConverter(&imageFormatConverter));
+
+  RETURN_NULL_IF_FAILED(imageFormatConverter->Initialize(
+      image->ImageSource().Get(), guid, WICBitmapDitherTypeNone, nullptr, 0.f,
+      WICBitmapPaletteTypeCustom));
+
+  size_t height, width;
+  RETURN_NULL_IF_FAILED(image->ImageSource()->GetSize(&width, &height));
+
+  const unsigned int size = height * width * 4;
+  woc::unique_iw<unsigned char> data(
+      static_cast<unsigned char *>(IwMalloc(size)));
+  RETURN_NULL_IF_FAILED(
+      imageFormatConverter->CopyPixels(0, width * 4, size, data.get()));
+
+  return
+      [NSData dataWithBytesNoCopy:data.release() length:size freeWhenDone:YES];
+}
+
+REFGUID _CGImageGetWICPixelFormat(unsigned int bitsPerComponent,
+                                  unsigned int bitsPerPixel,
+                                  CGColorSpaceRef colorSpace,
+                                  CGBitmapInfo bitmapInfo) {
+  CGColorSpaceModel colorSpaceModel =
+      ((__CGColorSpace *)colorSpace)->colorSpaceModel;
+
```

Comment:
CGColorSpaceGetModel. replaced.

---
In reply to: [84348935](https://github.com/Microsoft/WinObjC/pull/1185#discussion_r84348935) [](ancestors = 84348935)